File: src/NocWMS-Client/src/app/models/vendor.ts
import { Gender } from './gender';
export class Vendor {
    constructor(vendor?: Partial<Vendor>) {
        if (vendor) {
            this.id = vendor.id;
            this.vendorName = vendor.vendorName;
            this.gender = vendor.gender;
            this.address = vendor.address;
            this.contact = vendor.contact;
            this.duty = vendor.duty;
            this.fax = vendor.fax;
            this.email = vendor.email;
            this.postCode = vendor.postCode;
            this.createTime = vendor.createTime;
        }
    }

    public id: string;
    public vendorName: string;
    public gender: Gender;
    public address: string;
    public contact: string;
    public duty: string;
    public telephone: string;
    public fax: string;
    public postCode: string;
    public email: string;
    public remarks: string;
    public createTime: Date;
}

File: src/NocWMS-Server/Server.ApiService/Models/Vendor.cs
using System;
using System.Collections.Generic;
using KiraNet.Camellia.Infrastructure.DomainModel.Data;
using Server.ApiService.Common;

namespace Server.ApiService.Models
{
    public class Vendor : IEntity
    {
        public string Id { get; set; }
        public string VendorName { get; set; }
        public Gender Gender { get; set; } = Gender.Male;
        public string Address { get; set; }
        public string Contact { get; set; }
        public string Duty { get; set; }
        public string Telephone { get; set; }
        public string Fax { get; set; }
        public string PostCode { get; set; }
        public string Email { get; set; }
        public string Remarks { get; set; }
        public DateTime CreateTime { get; set; }

        public List<InboundReceipt> InboundReceipts { get; set; }
        public List<VendorProduct> VendorProducts { get; set; }
    }
}
